internal/ocm/addon_status_test.go

```diff
@@ -0,0 +1,238 @@
+package ocm
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+// TestClient_GetAddOnStatus verifies the behavior of the GetAddOnStatus method using
+// different scenarios.
+func TestClient_GetAddOnStatus(t *testing.T) {
+	// Create a mock HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Mock the response
+		w.WriteHeader(http.StatusOK)
+		_, err := w.Write([]byte(`{
+			"kind": "AddOnStatusResponse",
+			"addon_id": "addon-1",
+			"correlation_id": "12345-6789",
+			"status_conditions": []
+		}`))
+		require.NoError(t, err, "Error writing response")
+	}))
+	defer server.Close()
+
+	// Create a new client with the mock server's URL
+	client := &Client{
+		opts: ClientOptions{
+			Endpoint:  server.URL,
+			ClusterID: "1ou",
+		},
+		httpClient: http.DefaultClient,
```

Comment:
I think it's better to use the server client here https://pkg.go.dev/net/http/httptest#Server.Client